### ISSUE TITLE
marathon-lb won't start declining resource offers

### ISSUE THREAD

**Issue by juwi** [mesosphere/marathon-lb#316]
Hi,

I've just setup DCOS 1.8 and am trying to get marathon-lb up and running, however it declines all resource offers. After some digging around I found the following:
Request:
`ports=([Some(80), Some(443), Some(9090), Some(9091), Some(10000), Some(10001), Some(10002), Some(10003), Some(10004), Some(10005), Some(10006), Some(10007), Some(10008), Some(10009), Some(10010), Some(10011), Some(10012), Some(10013), Some(10014), Some(10015), Some(10016), Some(10017), Some(10018), Some(10019), Some(10020), Some(10021), Some(10022), Some(10023), Some(10024), Some(10025), Some(10026), Some(10027), Some(10028), Some(10029), Some(10030), Some(10031), Some(10032), Some(10033), Some(10034), Some(10035), Some(10036), Some(10037), Some(10038), Some(10039), Some(10040), Some(10041), Some(10042), Some(10043), Some(10044), Some(10045), Some(10046), Some(10047), Some(10048), Some(10049), Some(10050), Some(10051), Some(10052), Some(10053), Some(10054), Some(10055), Some(10056), Some(10057), Some(10058), Some(10059), Some(10060), Some(10061), Some(10062), Some(10063), Some(10064), Some(10065), Some(10066), Some(10067), Some(10068), Some(10069), Some(10070), Some(10071), Some(10072), Some(10073), Some(10074), Some(10075), Some(10076), Some(10077), Some(10078), Some(10079), Some(10080), Some(10081), Some(10082), Some(10083), Some(10084), Some(10085), Some(10086), Some(10087), Some(10088), Some(10089), Some(10090), Some(10091), Some(10092), Some(10093), Some(10094), Some(10095), Some(10096), Some(10097), Some(10098), Some(10099), Some(10100)] required + 105 dynamic)`

Offer:
`"ports" type: RANGES ranges { range { begin: 1025 end: 2180 } range { begin: 2182 end: 3887 } range { begin: 3889 end: 5049 } range { begin: 5052 end: 8079 } range { begin: 8082 end: 8180 } range { begin: 8182 end: 32000 } }`

So basically it cannot get ports 80 and 443. Sure enough those ports are not defined in MESOS_RESOURCES.

Is this a bug in DCOS or is it expected behaviour?
Also, what would be the correct way to expose those two additional ports without adding a whole range?

Thanks in advance!

**Comment by brndnmtthws** [comment]
The MLB package expects a public agent by default (denoted by the `slave_public` role). Public agents will have ports 80 and 443 available.

**Comment by juwi** [comment]
Yes, I figured out, that the file /var/lib/dcos/mesos-resources did not contain the correct properties for a public slave. I did however not check again if this was an error in the installation process (as in dcos put the wrong file there on install) or if this was due to the node being converted from a private to a public node.
